src/parser.c

```diff
@@ -3452,3 +3474,267 @@ void rbs_parser_set_error(rbs_parser_t *parser, rbs_token_t tok, bool syntax_err
   parser->error->message = message;
   parser->error->syntax_error = syntax_error;
 }
+
+/*
+  parse_method_overload ::= {} annotations <method_type>
+ */
+NODISCARD
+static bool parse_method_overload(rbs_parser_t *parser, rbs_node_list_t *annotations, rbs_method_type_t **method_type) {
+  rbs_position_t pos = NullPosition;
+
+  if (!parse_annotations(parser, annotations, &pos)) {
+    return false;
+  }
+
+  return rbs_parse_method_type(parser, method_type);
+}
+
+/*
+  inline_method_overloads ::= {} <overload>                        -- returns true
+                           | {} overload `|` ... `|` overload     -- returns true
+                           | {<>}                                 -- returns false
+*/
+NODISCARD
+static bool parse_inline_method_overloads(rbs_parser_t *parser, rbs_node_list_t *overloads, rbs_location_list_t *bar_locations) {
+  while (true) {
+    rbs_node_list_t *annotations = rbs_node_list_new(ALLOCATOR());
+    rbs_method_type_t *method_type = NULL;
+
+    if (!parse_method_overload(parser, annotations, &method_type)) {
+      return false;
+    }
+
+    rbs_location_t *location = rbs_location_new(ALLOCATOR(), parser->current_token.range);
+    rbs_ast_members_method_definition_overload_t *overload = rbs_ast_members_method_definition_overload_new(
+      ALLOCATOR(),
+      location,
+      annotations,
+      (rbs_node_t*)method_type
+    );
+    rbs_node_list_append(overloads, (rbs_node_t*)overload);
+
+    if (parser->next_token.type == pBAR) {
+      rbs_location_t *bar_location = rbs_location_new(ALLOCATOR(), parser->next_token.range);
+
+      rbs_parser_advance(parser);
+
+      rbs_location_list_append(bar_locations, bar_location);
+
+      continue;
+    }
+
+    return true;
+  }
+}
+
+NODISCARD
+static bool parse_inline_comment(rbs_parser_t *parser, rbs_location_t **comment) {
+  if (parser->next_token.type == tINLINECOMMENT) {
+    rbs_range_t comment_range = parser->next_token.range;
+    rbs_parser_advance(parser);
+
+    *comment = rbs_location_new(ALLOCATOR(), comment_range);
+    return true;
+  }
+
+  *comment = NULL;
+  return true;
+}
```

Comment:
Could we invert to use an early return?


```suggestion
static bool parse_inline_comment(rbs_parser_t *parser, rbs_location_t **comment) {
  if (parser->next_token.type != tINLINECOMMENT) {
    *comment = NULL;
    return true;
  }

  rbs_range_t comment_range = parser->next_token.range;
  rbs_parser_advance(parser);

  *comment = rbs_location_new(ALLOCATOR(), comment_range);
  return true;
}
```